tools/fork-network/src/cli.rs

```diff
@@ -237,30 +237,47 @@ impl ForkNetworkCommand {
         let head = store.get_ser::<Tip>(DBCol::BlockMisc, FINAL_HEAD_KEY)?.unwrap();
         let shard_layout = epoch_manager.get_shard_layout(&head.epoch_id)?;
         let all_shard_uids = shard_layout.get_shard_uids();
-        let fork_head = get_fork_head(&all_shard_uids, store.clone())?;
-        tracing::info!(?fork_head);
+        let num_shards = all_shard_uids.len();
+        // Flat state can be at different heights for different shards.
+        // That is fine, we'll simply lookup state root for each .
+        let fork_heads = get_fork_heads(&all_shard_uids, store.clone())?;
+        tracing::info!(?fork_heads);
 
         let chain =
             ChainStore::new(store.clone(), near_config.genesis.config.genesis_height, false);
 
-        let header = chain.get_block_header(&fork_head.hash)?;
-        let epoch_id = header.epoch_id();
-        let num_shards = shard_layout.num_shards();
-        let block_hash = fork_head.hash;
-        let block_height = fork_head.height + 1;
+        // Move flat storage to the max height for consistency across shards.
+        let (block_height, desired_block_hash) =
+            fork_heads.iter().map(|head| (head.height, head.hash)).max().unwrap();
```

Comment:
Sanity check: Are we guaranteed that all flat storage for all shards can be moved there?
Sanity check: What db are you using here? Is it the real db of a live or stopped node? Is it the flat state snapshot? Is it an ad-hoc snapshot? Are you not breaking any invariants here?

Reply:
> Are we guaranteed that all flat storage for all shards can be moved there?

Probably not if:
* different shards correspond to different shard layouts.
* Node is not tracking all shards and doesn't have FlatStateChanges corresponding  to the desired block for one or more shards.


Otherwise, yes. If a flat head is behind, it simply means the blocks ahead of it have no flat state changes.